src/useDebouncedCallback/useDebouncedCallback.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { DependencyList, useMemo, useRef } from 'react';
+import { useUnmountEffect } from '../useUnmountEffect/useUnmountEffect';
+
+export interface IDebouncedFunction<Args extends any[], This> {
+  (this: This, ...args: Args): void;
+}
+
+/**
+ * Makes passed function debounced, otherwise acts like `useCallback`.
+ *
+ * @param callback Function that will be debounced.
+ * @param delay Debounce delay.
+ * @param deps Dependencies list when to update callback.
+ * @param maxWait Maximum amount of milliseconds that function can be delayed
```

Comment:
I like Lodash's description of `maxWait`

`The maximum time func is allowed to be delayed before it's invoked.`